discretisedfield/tests/test_field.py

```diff
@@ -932,28 +932,41 @@ def test_eq():
     assert f5 == f7
 
 
-# TODO Hans
-def test_allclose():
-    p1 = (-5e-9, -5e-9, -5e-9)
-    p2 = (15e-9, 5e-9, 5e-9)
-    cell = (5e-9, 1e-9, 2.5e-9)
-    mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+@pytest.mark.parametrize("tol_value", [1e-10, 1e-5, 0.5, 2])
+@pytest.mark.parametrize("base_value", [0.5, 1, -1, 1e-9])
+def test_allclose_rtol(valid_mesh, nvdim, tol_value, base_value):
+    base_value = np.full(nvdim, base_value)
+    f1 = df.Field(valid_mesh, nvdim=nvdim, value=base_value)
+    f2 = df.Field(valid_mesh, nvdim=nvdim, value=base_value * (1 + tol_value * 0.9))
+    f3 = df.Field(valid_mesh, nvdim=nvdim, value=base_value * (1 + tol_value * 1.1))
+
+    assert f1.allclose(f2, rtol=tol_value, atol=0)
+    assert not f1.allclose(f3, rtol=tol_value, atol=0)
+
+
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+@pytest.mark.parametrize("tol_value", [1e-7, 1e-5, 0.5, 2])
+@pytest.mark.parametrize("base_value", [0, 0.5, 1, -1, 1e-9])
+def test_allclose_atol(valid_mesh, nvdim, tol_value, base_value):
+    base_value = np.full(nvdim, base_value)
+    f1 = df.Field(valid_mesh, nvdim=nvdim, value=base_value)
+    f2 = df.Field(valid_mesh, nvdim=nvdim, value=base_value + tol_value * 0.9)
+    f3 = df.Field(valid_mesh, nvdim=nvdim, value=base_value + tol_value * 1.1)
+
+    # Need an rtol to deal with floating point accuracy
+    rtol = (base_value - tol_value) * 1e-10
```

Comment:
I am assuming the tests fail due to floating point accuracy limitation if `rtol=0`?

Reply:
Yes you are correct. If I put `rtol=0` floating point accuracy meant that it was never equal. Hence I picked a value of `rtol` that scaled with the values